instrumentation/dubbo-rpc/src/main/java/brave/dubbo/rpc/TracingFilter.java

```diff
@@ -121,21 +121,25 @@ public void setRpcTracing(RpcTracing rpcTracing) {
       span = serverHandler.handleReceive(serverRequest);
     }
 
-    boolean deferFinish = false;
+    boolean isSynchronous = true;
     Scope scope = currentTraceContext.newScope(span.context());
     Result result = null;
     Throwable error = null;
     try {
       result = invoker.invoke(invocation);
       error = result.getException();
       Future<Object> future = rpcContext.getFuture(); // the case on async client invocation
-      if (future instanceof FutureAdapter) {
```

Comment:
before, this instanceof was also checking null. In practice, this is always a (dubbo) FutureAdapter